inst/include/dust/gpu/filter_state.hpp

```diff
@@ -44,7 +44,7 @@ class filter_trajectories_device : public filter_trajectories_host<real_type> {
                                this->history_value.size() * sizeof(real_type),
                                cudaHostRegisterDefault));
     CUDA_CALL(cudaHostRegister(this->history_order.data(),
-                               this->history_order.size() * sizeof(real_type),
+                               this->history_order.size() * sizeof(size_t),
```

Comment:
This is an old bug, causes crash when saving trajectories from models that use `float` as `real_type`